draft-ietf-mls-architecture.md

```diff
@@ -1816,46 +1817,45 @@ emitted credentials might be compromised.
 Details about how to verify the identity of a client depend on the MLS
 Credential type used. For example, cryptographic verification of credentials can
-be largely performed autonomously (e.g. without user interaction) by
+be largely performed autonomously (e.g., without user interaction) by
 the clients themselves for the `x509` Credential
-type. In contrast, when MLS clients use the `basic` Credential type, a larger
-degree of trust must be placed in a (likely) centralized authentication
-resource, or on out-of-band processes such as manual verification.
+type.
+
+In contrast, when MLS clients use the `basic` Credential type, then some
+other mechanism must be used to verify identities. For instance the Authentication
+Service could operate some sort of directory server to provide keys,
+or users could verify keys via an out-of-band mechanism.
 
 > **RECOMMENDATION:** Select the strongest MLS Credential type available among
 > the target members of an MLS group.
 
-If the AS is compromised, it could validate a (or generate a new) signature
-keypair for an attacker. Because a user can have many MLS clients running the
-MLS protocol, it possibly has many signature keypairs for multiple
-devices. These attacks could be very difficult to detect.
-
-> **RECOMMENDATION:** Provide a key transparency mechanism for the
-> Authentication Services to allow public verification of the credentials
-> authenticated by this service.
-
-Note that when a `basic` Credential is used, the Authentication Service also
-needs an out-of-band mechanism to verify the identity asserted in the
-Credential.
-
-In the case where an adversarial keypair is generated for a specific identity,
-an infrastructure without any transparency mechanism or out-of-band
-authentication mechanism could inject a malicious client into a group by
-impersonating a user. This is especially the case in large groups where the UI
-might not reflect all the changes back to the users.
+If the AS is compromised, it could validate a (or generate a new)
+signature keypair for an attacker. Because a user can have many MLS
+clients running the MLS protocol, it possibly has many signature
+keypairs for multiple devices. These attacks could be very difficult
+to detect, especially in large groups where the UI might not reflect
+all the changes back to the users. If the application participates in
+a key transparency mechanism in which it is possible to determine
+every key for a given user, then this then this would allow for the
+detection of surreptitiously created false credentials.
 
 > **RECOMMENDATION:** Make sure that MLS clients reflect all the membership
 > changes to the users as they happen. If a choice has to be made because the
-> number of notifications is too high, a public log should be maintained of the
+> number of notifications is too high, the client should provide a log of
 > state of the device so that the user can examine it.
```

Comment:
Hmmm..... ISTM that there are two potential things this might be talking about:

1. Changes to the set of clients in a group.
2. Changes to the set of clients for a user, whether in a group or not.

I assumed that this text was about the former because of the text "This is especially the case in large groups where the UI might not reflect all the changes back to the users." which is clearly about groups, not the set of clients for a user.

I agree that it would be fine to talk about a transparency log for user credentials, but then this needs a substantial rewrite.